pkg/apis/networking/types.go

```diff
@@ -48,30 +48,44 @@ const (
 
 // NetworkPolicySpec provides the specification of a NetworkPolicy
 type NetworkPolicySpec struct {
-	// Selects the pods to which this NetworkPolicy object applies. The array of
-	// ingress rules is applied to any pods selected by this field. Multiple network
-	// policies can select the same set of pods. In this case, the ingress rules for
-	// each are combined additively. This field is NOT optional and follows standard
-	// label selector semantics. An empty podSelector matches all pods in this
-	// namespace.
+	// Selects the pods to which this NetworkPolicy object applies.
+	// A network flow is bidirectional but initiated in one direction,
+	// from a source to a destination.
+	// The whole flow is allowed or not depending on whether the initiation is allowed.
+	// A network flow with a pod on either or both ends is allowed to be initiated
+	// if both (a) initiation from the source is allowed and (b) initiation to the
+	// destination is allowed.
+	//
+	// Initiation from a source is allowed if one of the following is true.
+	// - The source is not a pod.
+	// - The source is a pod and there is at least one NetworkPolicy whose `podSelector`
+	//   selects the pod, `policyTypes` includes "Egress",
+	//   and `egress` includes a rule that allows the initiation.
+	// - The source is a pod and there is no NetworkPolicy whose `podSelector` selects
+	//   the pod and whose `policyTypes` includes "Egress".
```

Comment:
The standard way of describing NetworkPolicy is that pods are either isolated or not isolated, and NetworkPolicy only applies to isolated pods. Pods become isolated for ingress by having an ingress NetworkPolicy select them, and pods become isolated for egress by having an egress NetworkPolicy select them.

Reply:
As you say, there are two directions of isolation.  A pod is _not_ simply "isolated" or not.

Reply:
Also, I think that in plain English, "isolated" is an absolute thing (no exceptions) whereas "restricted" is not absolute (there are some exceptions).  What we mean here is not an absolute but rather a matter of imposing some limits.